shell/browser/api/electron_api_base_window.cc

```diff
@@ -759,6 +759,14 @@ void BaseWindow::AddBrowserView(v8::Local<v8::Value> value) {
     auto get_that_view = browser_views_.find(browser_view->ID());
     if (get_that_view == browser_views_.end()) {
       if (browser_view->web_contents()) {
+        // If we're reparenting a BrowserView, ensure that it's detached from
+        // its previous owner window.
+        auto* owner_window = browser_view->web_contents()->owner_window();
+        if (owner_window && owner_window != window_.get()) {
+          owner_window->RemoveBrowserView(browser_view->view());
+          browser_view->web_contents()->SetOwnerWindow(nullptr);
+        }
```

Comment:
getting the owner window via the WebContents feels pretty indirect. Can we get the parent BrowserWindow from the BrowserView more directly?

Reply:
not at present - browserview doesn't currently have a concept of its immediate parent window so we'd be adding a getter for this specific case which seems like a net neutral change given that we already have this pre-existing way to get it 🤔